contracts/fee_currencies.go

```diff
@@ -46,10 +52,21 @@ func DebitFees(evm *vm.EVM, feeCurrency *common.Address, address common.Address,
 	// Hide this function from traces
 	if evm.Config.Tracer != nil && !evm.Config.Tracer.TraceDebitCredit {
 		origTracer := evm.Config.Tracer
+		evm.Config.Tracer = nil
+
+		var origHooks *tracing.Hooks
+		hookedStateDB, isHookedStateDB := evm.StateDB.(hookedStateDB)
+		if isHookedStateDB {
+			origHooks = hookedStateDB.Hooks()
+			hookedStateDB.SetHooks(nil)
+		}
+
 		defer func() {
 			evm.Config.Tracer = origTracer
+			if isHookedStateDB {
+				hookedStateDB.SetHooks(origHooks)
+			}
 		}()
-		evm.Config.Tracer = nil
 	}
```

Comment:
Since this is getting longer, would it make sense to extract it into a function?
Something like
```
reenableTracer := disableTracer(evm)
defer reenableTracer
```